### ISSUE TITLE
Doesn't see any files? Can't save changes.

### ISSUE THREAD

**Issue by RDKRACZ** [SpartanJ/ecode#99]
ecode starts in my home dir but the only folders it does see are ".local" and ".var".
No Documents and other home directories are detected. It even allows me to create Documents directory and then pretends to save a new file inside it but no changes are saved.

**Comment by SpartanJ** [comment]
If you go to open a folder you can't navigate it? Which OS are you using? Can you send me a screenshot or a video? You can open any folder just passing the folder path as a command line parameter too: `ecode /my/folder`.

**Comment by jozefcifre** [comment]
I'm experiencing the same problem on MX Linux Xfce (latest stable version), ecode 0.4.9 (Vajra) from Flathub. I can't see any folder other than ".local" and ".var", also, editing the path won't work, when I click the "Open" button it does nothing.
![image](https://github.com/SpartanJ/ecode/assets/124826525/5b23c106-d1fb-442a-9c60-83ae17d0dad0)
![image](https://github.com/SpartanJ/ecode/assets/124826525/679bc8f0-a4cc-4fc3-bba4-019d090f400a)

**Comment by SpartanJ** [comment]
Ohh, I see... it seems that flatpak uses some kind of virtual file system. IDK if this was working before... I'll take a look, but this is a packaging issue. Meanwhile please use the available downloads from this repository.

**Comment by jozefcifre** [comment]
> Ohh, I see... it seems that flatpak uses some kind of virtual file system. IDK if this was working before... I'll take a look, but this is a packaging issue. Meanwhile please use the available downloads from this repository.

Hm, that might be the proble, it does not integrate well with the host system.
I hope it gets solved.

**Comment by SpartanJ** [comment]
It can be solved, it's just a packaging issue. I was not responsible for creating the flatpak version so I never used it actively (I simply checked if the app was opening). 
[Here](https://docs.flatpak.org/en/latest/sandbox-permissions.html)'s the documentation for myself.
I tentatively should add to [this](https://github.com/flathub/dev.ensoft.ecode/blob/master/dev.ensoft.ecode.yml#L12) the following options:
```
--filesystem=host
--socket=pulseaudio
```

I'll test it this weekend.

**Comment by SpartanJ** [comment]
BTW, you can force the file system access, follow this [guide](https://davejansen.com/give-full-filesystem-access-to-flatpak-installed-applications/).

**Comment by jozefcifre** [comment]
Thanks for the info. By the way, I just downloaded the app image which doesn't have that problem, but I'd rather have it installed on my system so I can run commands from the terminal.

**Comment by SpartanJ** [comment]
I made a few changes that will give host file system access, but ecode also relies on external CLI applications that cannot be acceded with the flatpak isolation model, so it's a really bad way to distribute ecode (I was not familiar at all with flatpak, I don't use it, but it was a contribution from another user so I continued updating the build. To enable access to external applications the user will have to configure ecode CLI dependencies manually, it's far from ideal and breaks the simplicity of ecode, but I'll continue making it available in flathub because it gives some visibility to ecode. Currently I'm waiting [this PR](https://github.com/flathub/flatpak-builder-lint/pull/153) to be accepted in order to be able to provide a way to users to access external CLI apps. They will need to follow [some instructions](https://github.com/SpartanJ/dev.ensoft.ecode/blob/v0.4.92/ECODE_FIRST_RUN.md) that hope they help to guide the adventurous users. ecode [will warn](https://github.com/SpartanJ/eepp/commit/9814f25696f6136996550dcfd03c43f26fc75bd5#diff-7cd51b497f3b7e81d0dc33654aa0097ca446cab43a675440643fb6ffbe8721c8R3402) the user when running from flatpak on its first run that it's not the recommended user to use the editor and provided a link to the documentation mentioned.
